tutorials/parquet-catalog-demos/euclid-q1-hats/4-euclid-q1-hats-magnitudes.md

```diff
@@ -0,0 +1,357 @@
+
+The best-estimate flux in the detection band is given by the column `mer_flux_detection_total`.
+This can also be used to color-correct flux measurements in non-detection bands, as we will demonstrate.
+The object detection process is described in Romelli.
+The [MER Photometry Cookbook](http://st-dm.pages.euclid-sgs.uk/data-product-doc/dmq1/merdpd/merphotometrycookbook.html) describes the color corrections and how to convert from flux to magnitude.
+In this tutorial, we will restrict to objects detected in VIS (I band) because it simplifies the calculations.
+
++++
+
+## 2. Imports and Paths
+
+```{code-cell} ipython3
+# # Uncomment the next line to install dependencies if needed.
+# %pip install hpgeom matplotlib pandas pyarrow
+```
+
+```{code-cell} ipython3
+import hpgeom  # Find HEALPix indexes from RA and Dec
+import matplotlib.pyplot as plt  # Create figures
+import pyarrow.compute as pc  # Filter dataset
+import pyarrow.dataset  # Load the dataset
+import pyarrow.fs  # Simple S3 filesystem pointer
+import pyarrow.parquet  # Load the schema
+```
+
+```{code-cell} ipython3
+# AWS S3 paths.
+s3_bucket = "nasa-irsa-euclid-q1"
+dataset_prefix = "contributed/q1/merged_objects/hats/euclid_q1_merged_objects-hats/dataset"
+dataset_path = f"{s3_bucket}/{dataset_prefix}"
+
+# S3 pointer. Use `anonymous=True` to access without credentials.
+s3 = pyarrow.fs.S3FileSystem(anonymous=True)
+```
+
+## 3. Load Template-fit and Aperture Magnitudes
+
+The following columns will be important.
+Descriptions come from Romelli.
+
+```{code-cell} ipython3
+# Whether the source was detected in VIS mosaic (1) or only in NIR-stack mosaic (0).
+VIS_DET = "mer_vis_det"
+
+# Best estimate of the total flux in the detection band.
+# From aperture photometry within a Kron radius.
+# Detection band is VIS if `VIS_DET == 1`. Otherwise, this is a
+# non-physical NIR-stack flux and there was no VIS detection (aka, NIR-only).
+# We will only deal with VIS-detected objects in this notebook.
+FLUX_DET_TOTAL = "mer_flux_detection_total"
+
+# Peak surface brightness minus the magnitude used for `mer_point_like_prob`.
+# This is a measure of compactness.
+MUMAX_MINUS_MAG = "mer_mumax_minus_mag"
+
+# Whether the detection has a >50% probability of being spurious (1=Yes, 0=No).
+SPURIOUS_FLAG = "mer_spurious_flag"
+
+# PHZ classification: 1=Star, 2=Galaxy, 4=QSO.
+# Combinations (3, 5, 6, and 7) indicate multiple probability thresholds were exceeded.
+PHZ_CLASS = "phz_phz_classification"
+```
+
+We'll convert the catalog fluxes to magnitudes following the [MER Photometry Cookbook](http://st-dm.pages.euclid-sgs.uk/data-product-doc/dmq1/merdpd/merphotometrycookbook.html).
+For convenience, we'll have PyArrow do the conversion during the read operation and return only the magnitudes.
```

Comment:
We should add a sentence here something along the lines of: pyarrow.compute functions that you use in the code below are described here: https://arrow.apache.org/docs/python/api/compute.html

For reader, this is a new way of doing column math that's worth the mention